Domaci_ukol_Automatizovane_testovani_Cvetelina/src/main/java/com/cvetelina/book/HomePage.java

```diff
@@ -32,36 +36,38 @@ public void displayInputTest() {
   @Test
   public void findUrlsTest() {
     List<String> expectedNames = Arrays.asList("Chapter1", "Chapter2", "Chapter3", "Chapter4", "Chapter5");
+    System.out.println("Expected names:");
     System.out.println(expectedNames);
     List<WebElement> liElements = driver.findElements(By.xpath(CHAPTERS_LIST));
+    System.out.println("Actual names");
+    System.out.println(liElements);// This println() actually does't give me the strings of the elements, tried this liElements.forEach(System.out::println);
     for (int i = 0; i < expectedNames.size(); i++) {
-      Assert.assertEquals(liElements.get(i).getText(), expectedNames.get(i));
+      Assert.assertEquals(liElements.get(i).getText(), expectedNames.get(i), "Expected names do not match actual.");
     }
   }
 
   @Test
-  public void displayHeadingTest(){
-    String expectedHeading = "Selenium: Beginners Guide";
-    String actualHeading = driver.getTitle();
-    Assert.assertEquals(expectedHeading, actualHeading);
+  public void displayHeadingTest() {
+    WebElement expectedHeading = driver.findElement(By.className("mainheading"));
+    String actualHeading = expectedHeading.getText();
+    System.out.println(actualHeading);
```

Comment:
Not needed to print actual heading value here. It's already done by assertEquals under.
Consider something like: Assert.assertEquals("Selenium: Beginners Guide", actualHeading, "Expected and actual heading.

Reply:
Ok, thanks!